ocaml/auth/xa_auth_stubs.c

```diff
@@ -70,6 +71,30 @@ CAMLprim value stub_XA_mh_chpasswd(value username, value new_password){
     CAMLreturn(ret);
 }
 
+#include <crypt.h>
+/* 'constructor' attribute will ensure this function gets call early during program startup. */
+void __attribute__((constructor)) stub_XA_workaround(void)
```

Comment:
I don't see an ml file referring to this stub. What actually calls it?

Reply:
I was thinking about adding a stub and a call in the module initialization `let () =...` but this is simpler.
The constructor gets called automatically when the shared library or process is starting up, and at that point you haven't created any threads yet so there are no race conditions.